File: scala/src/test/scala/fun/pierre/lists/P05Test.scala
package fun.pierre.lists

import org.scalatest.{FunSpec, Matchers}

class P05Test extends FunSpec with Matchers {

  describe("P05 reverse element") {
    it("P05 reverse char element") {
      val k = P05.reverseElement(List("a", "b", "c", "d"))
      k should be(List("d", "c", "b", "a"))
    }

    it("P05 reverse int element") {
      val k = P05.reverseElement(List(1, 2, 3, 4, 5))
      k should be(List(5, 4, 3, 2, 1))
    }
  }
}

File: java/src/test/java/fun/pierre/lists/P05Test.java
package fun.pierre.lists;

import org.junit.Test;
import static java.util.Arrays.asList;
import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class P05Test {
    @Test
    public void shouldFindReverseElementOfIntList() {
        assertThat(P05.reverseElement(asList(1, 2, 3, 4, 5)), is(equalTo(asList(5, 4, 3, 2, 1))));
    }

    @Test
    public void shouldFindReverseElementOfCharList() {
        assertThat(P05.reverseElement(asList('a', 'b', 'c', 'd')), is(equalTo(asList('d', 'c', 'b', 'a'))));
    }
}
